### PR TITLE
feat: add ses smpt password to ssm, add ssm base path option

### LINKED ISSUE #60: Add ses_smtp_password_v4 to SSM
## Describe the Feature

It's really useful to store ses_smtp_password_v4 in SSM.

### PR DESCRIPTION
**Pull request by Gowiem**
## what
* Add option to change the SSM base path under which credentials are stored
* Store ses_smtp_password_v4 in SSM

## why
* Even if the IAM users are global per account often times there is a need to separeate them per environment (dev, staging). By adding the `ssm_base_path` parameter we can have a better SSM hierarchy. Example: `/dev/system_user/tc-dev-s3` and `/staging/system_user/tc-staging-ses`
* Also having the `ses_smtp_password_v4` stored in SSM will allow better integration with CI systems when deploying applications (rather than having a script to generate the password or read the terraform state)

## references
* Link to any supporting github issues or helpful documentation to add some context (e.g. stackoverflow). 
* closes #60
* Copy of #61, Closes #61

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -66,22 +66,29 @@ module "store_write" {
```
Comment (nitrocode) [inline main.tf:66]: Might as well bump the module to 0.9.1

**Comment by nitrocode** [comment]
@Gowiem conflict

**Comment by Gowiem** [comment]
@nitrocode have you seen those auto-readme conflicts before? Not sure what is up there... I believe I want to accept all the incoming changes, but wanted to get another set of eyes before I undo some work if I'm wrong.

**Review by Nuru — APPROVED** [review]

**Pull request merged**